packages/remix-dev/cli/migrate/migrations/replace-remix-imports/getTransformOptions.ts

```diff
@@ -1,30 +1,74 @@
+import inquirer from "inquirer";
 import type { PackageJson } from "type-fest";
 
+import { error, hint } from "../../../../logging";
 import type { Options } from "./transform";
-import { adapters, runtimes } from "./transform";
+import { runtimes, isRuntime, isAdapter } from "./transform";
 import type {
   Adapter,
   Runtime,
 } from "./transform/mapNormalizedImports/packageExports";
 
-const getAdapter = ({ dependencies }: PackageJson): Adapter =>
-  Object.keys(dependencies || {})
-    .filter(
-      (key) =>
-        key.startsWith("@remix-run/") &&
-        !["react", "serve", "server-runtime", ...runtimes]
-          .map((pkgName) => `@remix-run/${pkgName}`)
-          .includes(key)
-    )
-    .map((key) => key.replace("@remix-run/", ""))
-    .find((key) =>
-      adapters.includes(key as typeof adapters[number])
-    ) as Adapter;
-
-const getRuntime = ({ scripts }: PackageJson): Runtime =>
-  (scripts?.postinstall?.replace("remix setup ", "") as Runtime) || "node";
-
-export const getTransformOptions = (packageJson: PackageJson): Options => ({
-  adapter: getAdapter(packageJson),
-  runtime: getRuntime(packageJson),
+const resolveRuntime = async ({ scripts }: PackageJson): Promise<Runtime> => {
+  // match `remix setup <runtime>` in `postinstall` script
+  let remixSetupMatch = scripts?.postinstall?.match(/remix setup (\w+)/);
+  if (remixSetupMatch && remixSetupMatch.length >= 2) {
+    let postinstallRuntime = remixSetupMatch[1];
+    if (isRuntime(postinstallRuntime)) {
+      return postinstallRuntime;
+    }
+  }
+
+  // otherwise, ask user for runtime
+  let { runtime } = await inquirer.prompt<{ runtime?: Runtime }>([
+    {
+      name: "runtime",
+      message: "Which server runtime is this project using?",
+      type: "list",
+      pageSize: runtimes.length + 1,
+      choices: [...runtimes, { name: "Nevermind...", value: undefined }],
+    },
+  ]);
+  if (runtime === undefined) process.exit(0);
```

Comment:
This will make the code less error-prone when needing to change it

```suggestion
  if (runtime === undefined) {
    process.exit(0);
  }
```

Reply:
We have formatting/code style enforced by prettier/eslint, and we don't currently take a stance against this style. We have many other occurrences of this in the codebase (`grep -nri ") return" packages`).

If we decide we don't want this style at some point in the future, we will proceed by adding the corresponding formatting/linting rules in eslint and autofixing all occurrences in the codebase.